### PR TITLE
[SPARK-14712][ML] LogisticRegressionModel.toString should summarize model

### PR DESCRIPTION
**Pull request by jiayue-zhang**
## What changes were proposed in this pull request?

[SPARK-14712](https://issues.apache.org/jira/browse/SPARK-14712)
spark.mllib LogisticRegressionModel overrides toString to print a little model info. We should do the same in spark.ml and override repr in pyspark.

## How was this patch tested?

LogisticRegressionSuite.scala
Python doctest in pyspark.ml.classification.py

### CODE REVIEW

**Comment by bravo-zhang** [comment]
Hi @holdenk , I'm opening this PR to continue the effort in https://github.com/apache/spark/pull/12491
When adding doctest, I noticed that the `uid` part of the string is always a random UUID, for example `LogisticRegression_42a29165218fb4d4f81d22`, generated in Scala even after I call `_resetUid`. So I re-construct the `__repr__` string in Python code.
Really appreciate it if @yanboliang can also take a look.

**Comment by holdenk** [comment]
On the Python side using __repr__ looks reasonable (although I would like to see a doctest for this). But we really should get @jkbradley or maybe @dbtsai to take a look on the ML side.

Jenkins ok to test.

Sorry for the delay on getting to this.

**Comment by bravo-zhang** [comment]
This PR recently got tested so it draws my attention. Is this something we want to proceed? @holdenk @yanboliang @jkbradley @dbtsai I don't see how the test failures relate to LogisticRegression. Might it be some other flaky tests? Also tagging @HyukjinKwon who is active in maintaining stale PRs.

**Comment by HyukjinKwon** [comment]
@bravo-zhang, mind if I ask to rebase it and see if the tests pass? BTW, let's fix the PR title to link the JIRA.

**Comment by bravo-zhang** [comment]
@HyukjinKwon It's ready to test.

**Comment by bravo-zhang** [comment]
@HyukjinKwon Can you recommend someone to take a look at this PR or maybe you can take a look?

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by HyukjinKwon** [comment]
I think you cc'ed right ones ..

File: python/pyspark/ml/classification.py [hunk 1]
```diff
@@ -562,6 +564,11 @@ def evaluate(self, dataset):
         java_blr_summary = self._call_java("evaluate", dataset)
         return BinaryLogisticRegressionSummary(java_blr_summary)
 
+    def __repr__(self):
```
Comment (holdenk) [inline python/pyspark/ml/classification.py]: So my question here is why we aren't calling the Java/Scala toString method directly as we do in the mllib one and in many of the other models in `regression.py` for the ml one?

**Review by holdenk — COMMENTED** [review]
Sorry for not noticing this earlier, looks really good - quick question on the Python side though. Would love to get this in :)

**Comment by bravo-zhang** [comment]
@holdenk @HyukjinKwon I updated `__repr__` by calling `toString`. I also added class name `LogisticRegressionModel: ` to `toString`. Otherwise `uid` alone is a bit confusing.

**Comment by holdenk** [comment]
LGTM merging to master.

**Comment by holdenk** [comment]
Thanks for the improvement :)

**Pull request closed without merging**